src/main/java/com/iexec/sms/tee/session/base/SecretSessionBaseService.java

```diff
@@ -159,6 +202,11 @@ SecretEnclaveBase getPreComputeTokens(final TeeSessionRequest request) throws Te
                                 // extract <IEXEC_INPUT_FILE_URL_N, url>
                                 || e.getKey().startsWith(IexecEnvUtils.IEXEC_INPUT_FILE_URL_PREFIX))
                 .forEach(e -> tokens.put(e.getKey(), e.getValue()));
+
+        // enclave signature
+        final Map<String, String> signTokens = getSignTokens(request, PRE_COMPUTE_STAGE);
+        tokens.putAll(signTokens);
```

Comment:
In those cases, we don't need to create the variable
```suggestion
        tokens.putAll(getSignTokens(request));
```

Or for a more potimized implementation, should we call getSignTokens from getSecretsTokens and inject its result in getPreComputeTokens and getPostComputeTokens ?

Reply:
Resolved in [dbfd39b](https://github.com/iExecBlockchainComputing/iexec-sms/pull/296/commits/dbfd39b146d7612e20bec9727f9de6860694d2dc)